### ISSUE TITLE
Compiling performances

### ISSUE THREAD

**Issue by kevin-lesterius** [symfony/webpack-encore#198]
Hi

I just tried out Webpack Encore and everything is working well but performances during the compilation are really bad to me.

I was using a gulp file before and watch/compile was about some ms and now when I change something watch/compile with webpack is about 5-8s. 

What am I doing wrong ?

```javascript
let Encore                = require('@symfony/webpack-encore');
let webpack               = require('webpack');
let UglifyJsPlugin        = require('uglifyjs-webpack-plugin');
let FaviconsWebpackPlugin = require('favicons-webpack-plugin');

Encore
    .setOutputPath('web/build/')
    .setPublicPath('/build')
    .cleanupOutputBeforeBuild()
    .addEntry('app', './app/Resources/assets/js/main.js')
    .createSharedEntry('vendor', [
        'jquery',
        'foundation-sites',
        'foundation-sites/dist/css/foundation.css'
    ])
    .enableSassLoader()
    .enablePostCssLoader()
    .autoProvidejQuery()
    .enableSourceMaps(!Encore.isProduction())
    .enableVersioning()
    .addPlugin(new FaviconsWebpackPlugin({
        logo: './app/Resources/assets/master-favicon.png',
        prefix: 'favicons-[hash]/',
        emitStats: true,
        statsFilename: 'favicons.json',
        persistentCache: true
    }))
;

let webpackConfig = Encore.getWebpackConfig();
// Use polling instead of inotify because of Vagrant
webpackConfig.watchOptions = { poll: true, ignored: /node_modules/ };

// Fix for UglifyJs (loaded 1.0.0-beta)
webpackConfig.plugins = webpackConfig.plugins.filter(
    plugin => !(plugin instanceof webpack.optimize.UglifyJsPlugin)
);
webpackConfig.plugins.push(new UglifyJsPlugin());


module.exports = webpackConfig;
```

This is an exemple of my watch performance without the favicons (First compilation is longer with favicons but that's another issue).
```
yarn run v1.2.1
$ yarn run encore dev --watch
Running webpack ...


Webpack is watching the files…

 DONE  Compiled successfully in 13334ms                                                                                                                                                                                                                                                                                                               

 I  6 files written to web/build
 WAIT  Compiling...        
                                                                                                                                                                                                                                                                                                                          
 DONE  Compiled successfully in 6322ms                                                                                                                                                                                                                                                                                                                

 I  6 files written to web/build
 WAIT  Compiling...           
                                                                                                                                                                                                                                                                                                                        
 DONE  Compiled successfully in 6023ms  

 I  6 files written to web/build
```

**Comment by stof** [comment]
I see one thing in your config already: you register an UglifyJsPlugin unconditionally. this means that you spend time minifying the JS in dev mode too. This is not what Encore is doing by default.

In my own project (not yet based on webpack, but still on gulp), I know that enabling minification makes my build 18x longer (minifying and mangling takes time).

So make sure that you don't apply minification in dev mode (your watch/compile cycle is in dev mode)

**Comment by kevin-lesterius** [comment]
Thanks it's a lot faster 😄 

I still have a problem because we're using a vagrant and sometimes it takes up to 5s to detect the changes in files. Is it related to Encore ? I guess thats more about our Vagrant configuration or Webpack itself ?

**Comment by Lyrkan** [comment]
It's probably due to the default poll interval used by Webpack (since you use `poll: true`).

Based on [their doc](https://webpack.js.org/configuration/watch/#watchoptions-poll) you can specify a different poll interval, maybe you could try something like `poll: 1000`.

**Comment by kevin-lesterius** [comment]
Hi @Lyrkan 
I already tried poll: 100 but that still takes 5s.
Don't you think it's more related to the sync files between local & vagrant ?

**Comment by stof** [comment]
@kevin-lesterius it might be related, indeed (as the polling will see the new file only after the sync is done)

**Comment by stof** [comment]
default poll interval when passing `true` is already `100`, so 5s is likely to be the vagrant syncing

**Comment by kevin-lesterius** [comment]
Ok I'll have a look further in this way, thank you

**Comment by scuben** [comment]
@kevin-lesterius try set `:mount_options => ['actimeo=1'],` in your `nfsoptions`. i have it solved with:

```
nfsoptions = {
            :create => true,
            :mount_options => ['actimeo=1'],
            :nfs => true,
            :nfs_udp => setupConfig['nfsoptions']['udp'],
            :bsd__nfs_options => setupConfig['nfsoptions']['bsd'],
            :linux__nfs_options => setupConfig['nfsoptions']['linux']
        }
```

**Comment by kevin-lesterius** [comment]
@scuben Hi, thank you, we already used this configuration on another project but here the Vagrant configuration is managed by a Puphpet file and I can't find how to set this option in a puphpet/config.yaml file
